dashboard/run_dashboard_app.py

```diff
@@ -229,14 +369,50 @@ def get_daily_user_chours(hpc: const.HPC, users_dict: Dict[str, str] = USERS):
     return data
 
 
-def get_chours_entries(hpc: const.HPC):
+def get_total_user_chours(
+    hpc: const.HPC,
+    users_dict: Dict[str, str] = USERS,
+    start_date=DEFAULT_START,
+    end_date=DEFAULT_END,
+):
+    """Get total core hours usage for a list of users in a specified period
+       Return as a table
+    """
+    data = []
+    for username, real_name in users_dict.items():
+        name, total_chours = app.db.get_total_user_chours(
+            hpc, username, start_date, end_date
+        )
+        data.append({"username": name, "total_core_hours": total_chours})
+    return html.Div(
+        [
+            dash_table.DataTable(
+                id="table",
+                columns=[
+                    {"id": c, "name": c} for c in ["username", "total_core_hours"]
+                ],
+                data=sorted(data, key=lambda k: k["total_core_hours"], reverse=True),
+                style_cell={"textAlign": "left"},
+            )
+        ],
+        id="table_container",
+        style={"float": "centre", "width": "50%", "padding-left": 50},
+    )
+
+
+def get_chours_entries(hpc: const.HPC, start_date=DEFAULT_START, end_date=DEFAULT_END):
     """Gets the core hours entries for the specified HPC
     Note: Only maui is currently supported
     """
     # Get data points
-    entries = app.db.get_chours_usage(
-        date.today() - relativedelta(days=188), date.today(), hpc
-    )
+    entries = app.db.get_chours_usage(start_date, end_date, hpc)
+    # reset start of a new allocation to 0 hours
+    start_total = entries[0][-1]
+    for i in range(len(entries)):
```

Comment:
entries = [(*entry[:2], entry[3] - start_total) for entry in entries] 
(just double check that this is actually correct..)

Reply:
Updated